bootstrap-salt.sh

```diff
@@ -638,7 +638,8 @@ fi
 
 # Root permissions are required to run this script
 if [ "$($whoami)" != "root" ]; then
-    echoerror "Salt requires root privileges to install. Please re-run this script as root."
+    echowarn "Salt requires root privileges to install. Re-running this script as root..."
+    exec "su -l root -c $0"
```

Comment:
What if only sudo or doas are available?

Reply:
I guess it could go through a series of checks to see which one will work. I'm not knowledgeable enough about what common AMIs have installed on ec2 or for that matter on a range of other cloud platforms. Its not possible to have a FreeBSD system without su. What's the situation commonly on Linux?

This solves a particular problem in that its impossible to get FreeBSD to easily work with default salt-cloud options since it just assumes sudo is always available on all platforms.

Reply:
At any rate, this patch doesn't have to be perfect. Right now the script just fails if it isn't root. So failing only sometimes is an improvement.